src/zjs_callbacks.c

```diff
@@ -313,19 +313,9 @@ void zjs_remove_callback(int32_t id)
 
 void zjs_signal_callback(int32_t id, void* args, uint32_t size)
 {
-    if (args && size) {
-        // signal with args
-        int ret = zjs_port_ring_buf_put(&ring_buffer, (uint16_t)id, 0, (uint32_t*)args, (uint8_t)size / 4);
-        if (ret != 0) {
-            DBG_PRINT("error putting into ring buffer, ret=%u\n", ret);
-        }
-    } else {
-        // signal without args
-        uint32_t dummy = 0;
-        int ret = zjs_port_ring_buf_put(&ring_buffer, (uint16_t)id, 0, (uint32_t*)&dummy, (uint8_t)1);
-        if (ret != 0) {
-            DBG_PRINT("error putting into ring buffer, ret=%u\n", ret);
-        }
+    int ret = zjs_port_ring_buf_put(&ring_buffer, (uint16_t)id, 0, (uint32_t*)args, (uint8_t)((size + 3) / 4));
+    if (ret != 0) {
+        DBG_PRINT("error putting into ring buffer, ret=%u\n", ret);
```

Comment:
Sorry, I hate to keep making comments... but I think this needs to be a real error, not just debug. If we weren't able to put then the callback is lost, and the user needs to have an indication (at least if they're watching serial) that their queue is full and may need to be increased or their algorithm throttle work.